pkg/reconciler/shared/tektonconfig/pipeline/pipeline.go

```diff
@@ -20,108 +20,104 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"log"
 	"reflect"
 
 	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
 
 	op "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
-	operatorv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
 	"github.com/tektoncd/operator/pkg/reconciler/common"
 	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"knative.dev/pkg/test/logging"
 )
 
-func CreatePipelineCR(ctx context.Context, instance v1alpha1.TektonComponent, client operatorv1alpha1.OperatorV1alpha1Interface) error {
-	configInstance := instance.(*v1alpha1.TektonConfig)
-	if _, err := ensureTektonPipelineExists(ctx, client.TektonPipelines(), configInstance); err != nil {
-		return errors.New(err.Error())
-	}
-	if _, err := waitForTektonPipelineState(ctx, client.TektonPipelines(), v1alpha1.PipelineResourceName,
-		isTektonPipelineReady); err != nil {
-		log.Println("TektonPipeline is not in ready state: ", err)
-		return err
-	}
-	return nil
-}
-
-func ensureTektonPipelineExists(ctx context.Context, clients op.TektonPipelineInterface, config *v1alpha1.TektonConfig) (*v1alpha1.TektonPipeline, error) {
+func EnsureTektonPipelineExists(ctx context.Context, clients op.TektonPipelineInterface, config *v1alpha1.TektonConfig) (*v1alpha1.TektonPipeline, error) {
 	tpCR, err := GetPipeline(ctx, clients, v1alpha1.PipelineResourceName)
-	if err == nil {
-		// if the pipeline spec is changed then update the instance
-		updated := false
-
-		if config.Spec.TargetNamespace != tpCR.Spec.TargetNamespace {
-			tpCR.Spec.TargetNamespace = config.Spec.TargetNamespace
-			updated = true
-		}
-
-		if !reflect.DeepEqual(tpCR.Spec.Pipeline, config.Spec.Pipeline) {
-			tpCR.Spec.Pipeline = config.Spec.Pipeline
-			updated = true
-		}
-
-		if !reflect.DeepEqual(tpCR.Spec.Config, config.Spec.Config) {
-			tpCR.Spec.Config = config.Spec.Config
-			updated = true
-		}
 
-		if tpCR.ObjectMeta.OwnerReferences == nil {
-			ownerRef := *metav1.NewControllerRef(config, config.GroupVersionKind())
-			tpCR.ObjectMeta.OwnerReferences = []metav1.OwnerReference{ownerRef}
-			updated = true
-		}
-
-		if updated {
-			return clients.Update(ctx, tpCR, metav1.UpdateOptions{})
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			tpCR, err = CreatePipeline(ctx, clients, config)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			return nil, v1alpha1.RECONCILE_AGAIN_ERR
 		}
```

Comment:
```suggestion
	if err != nil {
		if apierrs.IsNotFound(err) {
			tpCR, err = CreatePipeline(ctx, clients, config)
			if err != nil {
				return nil, err
			}
			return nil, v1alpha1.RECONCILE_AGAIN_ERR
		} 
		return nil, err
```

Reply:
if any of the calls return an actual error, we return it as it is. When there is no error from Create call, we return a `Reconcile_Again` to make the reconciler come back.